### ISSUE TITLE
Use Orchid to maintain documentation and landing page site

### ISSUE THREAD

**Issue by cjbrooks12** [PebbleTemplates/pebble#337]
Orchid is a tool I recently released which is perfect for managing the documentation of Java projects, and Pebble is the ideal candidate. Orchid itself uses Pebble as its primary template engine, so setting up Pebble with Orchid docs not only will improve the documentation as a whole, but also serve as a great example of what can be done with Pebble.

This issue will track my progress getting Pebble set up with Orchid, and also serve as a discussion around the site.

**Progress**

- [ ] Landing page(s)
- [ ] Full wiki, migrated from the current Github Wiki
- [ ] Auto-generated listings of/documentation for:
    - [ ] Tags
    - [ ] Filters
    - [ ] Functions
    - [ ] Tests
    - [ ] Operators
- [ ] Complete Javadoc API documentation
- [ ] Automatic build/deploy from Travis CI to Github Pages

**Discussion Points**

- Orchid runs as a Gradle plugin, so using Orchid will require Pebble to either migrate from Maven to Gradle as its sole build system, or use both in the same project (Maven for normal build/test, Gradle for Orchid build). Alternatively, I would like to build a Maven plugin eventually but don't know how; if anyone has experience/would like to help out building a Maven Orchid plugin, we could continue using only Maven here
- What should the site look like? Keep the theme used for the current docs, use a default Orchid theme, or find/create a new theme just for Pebble

**Comment by cjbrooks12** [comment]
Alright, I've got the initial groundwork laid out for this on the [`docs` branch of my fork](https://github.com/JavaEden/pebble), and have a preview of the site deployed to https://pebbletemplates.netlify.com/ . Not much of the actual content has been migrated yet, but I've got it generating empty pages for everything registered in the CoreExtension as a start, currently, along with Javadocs.

**Comment by ebussieres** [comment]
@cjbrooks12 I'd prefer to stick with Maven for pebble project. I know it really well vs gradle.

**Comment by bjansen** [comment]
I agree that an actual documentation website would be easier to use than the current wiki where the menu is not user friendly.

Any news on what you started @cjbrooks12 ?

**Comment by cjbrooks12** [comment]
Sorry for the delay with this, there were a number of bugfixes and features Orchid needed before it could really do well as a great project documentation tool. But I have just gotten a few of my other small projects set up fully with Orchid docs, along with [Strikt](https://github.com/robfletcher/strikt), and feel like it's stable enough and now has the features needed to get Pebble set up with what it needs. I'll get these remaining tasks finished out soon (gotta [get those 5 PRs!](https://hacktoberfest.digitalocean.com/)).

**Comment by DirkLachowski** [comment]
@cjbrooks12 Looking at the strict docs they seem to be build using freemarker, not pebble?

**Comment by cjbrooks12** [comment]
I've been working with the maintainer to replace JBake with Orchid, [the PR is still open](https://github.com/robfletcher/strikt/pull/119)

**Comment by bjansen** [comment]
@cjbrooks12 is there something I could do to help you migrate the docs to Orchid? (gotta get those 5 PRs too :))

**Comment by cjbrooks12** [comment]
Do you know how to [make Maven plugins](https://github.com/JavaEden/Orchid/issues/100)? My current plan is to use a kscript [like this](https://github.com/copper-leaf/copper-leaf.github.io/blob/master/orchid.kts) to [package everything up](https://github.com/holgerbrandl/kscript#deploy-scripts-as-standalone-binaries) so that it doesn't need Gradle to run, but it would be nicer to have a Maven plugin which can run Orchid as part of the normal Pebble deploy process.

**Comment by bjansen** [comment]
Yes, I have written a couple plugins at work, I'll give it a try :)

**Comment by hectorlf** [comment]
Very cool!! Maybe add a footer? (at least there's none in the mobile layout)

**Comment by bjansen** [comment]
The landing page could be a little bit improved, especially if it becomes the landing page of pebbletemplates.io, but it's not really related to Orchid nor to this PR :) This can be done later.

**Comment by cjbrooks12** [comment]
Yeah I just used the current Wiki's landing page as the Orchid site homepage as well. And this theme is set up so the "footer" is at the bottom of the sidebar (which becomes a drawer on mobile). But yeah, the copyright on it should be updated to PebbleTemplates, is there anything else that should go in it?

Also, now that I'm thinking about it, it would be nice to have a contributor list in the site, and maybe changelogs as well. I'm not sure how the current release process goes and if it automatically creates a Github Release, but Orchid's [changelog plugin](https://orchid.netlify.com/plugins/OrchidChangelog) can create the release here at the same time the site is deployed, in addition to showing the full list of releases. Less manual work required for releasing new versions of Pebble 😉

**Comment by ebussieres** [comment]
The release process doesn't generate a Github release. I run mvn release:prepare release:perform which build everything and push it to sonatype. After that, I manually create the Github release and update documentation with the new version.

It would be awesome to integrate the github release based on a Changelog to the release process and i'm open to hear about it

**Comment by ebussieres** [comment]
@cjbrooks12 Can I close this issue ? I see that there's still an uncompleted task.

**Comment by bjansen** [comment]
Perhaps now http://pebbletemplates.io could show the new site instead of redirecting to the github project?

**Comment by ebussieres** [comment]
Yeah on next release
